osu.Game/Overlays/Profile/Sections/Recent/DrawableRecentActivity.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+using osu.Game.Screens.Select.Leaderboards;
+
+namespace osu.Game.Overlays.Profile.Sections.Recent
+{
+    public class DrawableRecentActivity : DrawableProfileRow
+    {
+        private APIAccess api;
+
+        private readonly RecentActivity activity;
+
+        private string userLinkTemplate;
+        private string beatmapLinkTemplate;
+        private string beatmapsetLinkTemplate;
+
+        private LinkFlowContainer content;
+
+        public DrawableRecentActivity(RecentActivity activity)
+        {
+            this.activity = activity;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(APIAccess api)
+        {
+            this.api = api;
+
+            userLinkTemplate = $"[{toAbsoluteUrl(activity.User?.Url)} {activity.User?.Username}]";

```

Comment:
i'd probably move these inside `activityToString()` as local functions rather than assigning them all in BDL (doesn't seem to be any reason for doing it this early).

i'd also consider moving the `MessageFormatter.FormatText` call in that method and rename it to `createMessage()` instead of the awkward `activityToString()`.